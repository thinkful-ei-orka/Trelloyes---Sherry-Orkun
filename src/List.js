import React from 'react';
import Card from './Card.js'

function List(props) {
    return (
      <section className="List">
          <header className="List-header">
            <h2>{props.header}</h2>
            <button onClick={() => props.create(props.id)}>Create a Card</button>
            {props.cards.map(card => <Card 
              delete={props.delete}
              title={card.title} 
              content={card.content} 
              key={card.id}
              id={card.id}/>)}
          </header>  
      </section>
    )
}

export default List;