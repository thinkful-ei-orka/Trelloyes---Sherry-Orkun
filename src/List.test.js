import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import List from './List';

 describe('<List />', () => {
   it('renders without crashing', () => {
     const section = document.createElement('section');

     ReactDOM.render(
       <List 
       header="Header"
       cards={[
       ]}
       />,
       section
     );
     
     ReactDOM.unmountComponentAtNode(section);
   });

   it('renders this UI as expected', () => {
    const tree = renderer.create(<List 
      header="Header"
      cards={[
      ]}
      />).toJSON();

    expect(tree).toMatchSnapshot();
  });
 })