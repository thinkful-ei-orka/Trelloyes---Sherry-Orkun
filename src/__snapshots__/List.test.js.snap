// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders this UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Header
    </h2>
  </header>
</section>
`;
